### ISSUE TITLE
[package] libsodium/1.0.18: compilation fails for x86

### ISSUE THREAD

**Issue by afedechko** [conan-io/conan-center-index#2992]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **libsodium/1.0.18**
  * Operating System+version: **Windows 10**
  * Compiler+version: **visual studio 17 (toolset v141)**
  * Conan version: **conan 1.25.2**

### Conan profile
```
Configuration:
[settings]
arch=x86
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.cppstd=17
compiler.runtime=MD
compiler.toolset=v141
compiler.version=15
os=Windows
os_build=Windows
[options]
[build_requires]
[env]
```

### Steps to reproduce (Include if Applicable)
have next packages within conanfiles.txt:
```
[requires]
boost/1.69.0@conan/stable
libcbor/0.7.0
cppzmq/4.6.0@bincrafters/stable
gtest/1.10.0

[generators]
cmake
```

install connan packages using 
```
conan install . -if build_x86 -s build_type=Release -s arch=x86 -s compiler.toolset=v141 -s compiler.cppstd=17 --build missing -g visual_studio_multi -g cmake_multi 
```
see error below

### Log
<details><summary>Click to expand log</summary>

```
red version of Windows SDK or change the SDK version in the project property pages or by right-clicking the solution and selecting "Retarget solution". [C:\Users\username\.conan\data\libsodium\1.0.18\_\_\build\e257c725700a61b4c6f028ef968079724fb2c75d\source_subfolder\builds\msvc\vs2017\libsodium\libsodium.vcxproj]
libsodium/1.0.18:
libsodium/1.0.18: ERROR: Package 'e257c725700a61b4c6f028ef968079724fb2c75d' build failed
libsodium/1.0.18: WARN: Build folder C:\Users\username\.conan\data\libsodium\1.0.18\_\_\build\e257c725700a61b4c6f028ef968079724fb2c75d
ERROR: libsodium/1.0.18: Error in build() method, line 159
        self._build_visual()
while calling '_build_visual', line 83
        msbuild.build(sln_path, upgrade_project=False, platforms={"x86": "Win32"}, build_type=self._vs_configuration)
        ConanException: Error 1 while executing set "VSCMD_START_DIR=%CD%" && call "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC/Auxiliary/Build/vcvarsall.bat" x86 && msbuild "C:\Users\username\.conan\data\libsodium\1.0.18\_\_\build\e257c725700a61b4c6f028ef968079724fb2c75d\source_subfolder\builds\msvc\vs2017\libsodium.sln" /p:Configuration="StaticRelease" /p:UseEnv=true /p:Platform="Win32" /m:8 /p:PlatformToolset="v141" /verbosity:minimal /p:ForceImportBeforeCppTargets="C:\Users\username\.conan\data\libsodium\1.0.18\_\_\build\e257c725700a61b4c6f028ef968079724fb2c75d\conan_build.props"
```

</details>

When I navigate to the place where the conan downloads packages, and tried to open libsodium.sln I see next message on VS:
```
C:\Users\username\.conan\data\libsodium\1.0.18\_\_\build\e257c725700a61b4c6f028ef968079724fb2c75d\source_subfolder\libsodium.vcxproj : warning  : The build tools for Visual Studio 2015 (v140) cannot be found. To build using the Visual Studio 2017 (v141) build tools, either click the Project menu or right-click the solution, and then select "Retarget Solution". Install Visual Studio 2015 (v140) to build using the Visual Studio 2015 (v140) build tools.
```
In short - for some reason it tries to use toolset v140 when I specify v141 explicitly in conan install command 

However, when I try to install conan dependencies for x64 architecture - all works as expected.

**Comment by Croydon** [comment]
Please migrate from `boost/1.69.0@conan/stable` to `boost/1.69.0` and from `cppzmq/4.6.0@bincrafters/stable` to `cppzmq/4.6.0` and check if you encounter the same problem

**Comment by afedechko** [comment]
Hello @Croydon, thank you for the reply.
Have just tried with the changes you suggested and still see the same behaviour. 
BTW are there any specific case when I should you `bincrafters/stable` or `conan/stable` ?

**Comment by Croydon** [comment]
`@username/channel` is optional for a while now.

All packages in the Conan Center Index don't have a `@username/channel` anymore. Packages with a `@username/channel` do come from places like [conan-community](https://github.com/conan-community) (`@conan)`) and [Bincrafters](https://github.com/bincrafters) (`@bincrafters)`) .

Generally, if a package is both in CCI and in Conan-Community or Bincrafters, only the package in CCI is still maintained.  Meaning, you should always prefer packages without `@username/channel`

**Comment by afedechko** [comment]
@Croydon, thank you for the explanation

**Comment by madebr** [comment]
@afedechko 
Is there still a problem after migrating to `cppzmq/4.6.0`, `boost/1.69.0` (or `boost/1.74.0`)?
Your requirements should then include `libsodium/1.0.18`.

**Comment by afedechko** [comment]
Yes, the problem still exists

**Comment by madebr** [comment]
@afedechko 
If you manually open `source_subfolder\builds\msvc\vs2017\libsodium\libsodium.vcxproj` with Visual Studio 2017 and retarget the solution yourself. Does it compile then? (right-click on project + retarget solution)

I have a similar problem with some projects.
Have you, now or in the past, installed a previous version of Visual Studio on this computer?
A computer of mine had in the past installed MSVC2013 and MSVC2015 on it.
Now it sometimes throws an error about WINDOWS SDK 10/8.1 retargeting.

**Comment by afedechko** [comment]
@madebr
As soon as I retarget the project it is compiled without problems. 

I had both Visual Studio 2015 and Visual Studio 2017 installed earlier. That is why I was not facing with the problem initially. However, sometime later I removed VIsual Studio 2015 and left only 2017. Started from then I see the problem.

**Comment by madebr** [comment]
You're having a similar problem as me.
There is something wrong with our configuration.
There's nothing conan or conan-center-index can do about this.
I found nothing when googling my problem.
If you end up finding/fixing your problem, please let me know :smile:

**Comment by afedechko** [comment]
The problem might be in the package itself, in the process that generates the project/solution.
I have bunch on other packages taken from conan: boost, libcbor, zmq, ..., an none of them has this problem only the libsodium one. 

Anyway, for now I have workaround for this - I removed the encryption for zmq, that in turn removed a need in libsodium.
